index-import.cpp

```diff
@@ -124,6 +142,13 @@ static IndexUnitWriter remapUnit(const std::unique_ptr<IndexUnitReader> &reader,
       fileMgr.getFile(mainFilePath), reader->isSystemUnit(),
       reader->isModuleUnit(), reader->isDebugCompilation(), reader->getTarget(),
       sysrootPath, moduleNames.getModuleInfo);
+  
+  // Check if the unit file is already up to date
+  SmallString<256> outputFileFullPath;
+  path::append(outputFileFullPath, workingDir, outputFile);
```

Comment:
@kastiglione, thanks for your comment!

> Is `outputFile` guaranteed to be relative? I think it can be absolute, I'll have to check.
You're right, it can be absolute. I'll update this part.

>For our builds, we don't copy the .o files, but we remap because that's what Xcode uses to lookup records in the index.
We need to use the remapped output file even though we don't copy them over. The purpose of the remapped output file path here is for calculating the Unit file name. Even though the ".o" file doesn't exist, we still need to calculate the unit file path based on it. Because as you mentioned, that's what Xcode uses to lookup the Unit file.

Reply:
I'll rename `outputFileFullPath` to `remappedOutputFilePath` to make it clearer.

Reply:
Got it. Thanks. Carry on :)